rest/blip_client_test.go

```diff
@@ -1332,6 +1208,7 @@ func (btr *BlipTesterReplicator) sendMsg(msg *blip.Message) {
 func (btcc *BlipTesterCollectionClient) upsertDoc(docID string, parentVersion *DocVersion, body []byte) *clientDocRev {
 	btcc.seqLock.Lock()
 	defer btcc.seqLock.Unlock()
+	newSeq := btcc._nextSequence()
```

Comment:
I think "allocating" a next sequence right before its used is safer in case we add any early returns/exits in this code.

I don't think it would cause functional issues but it would at least avoid unused allocated sequences in the client - I'm not sure whether the code is set up to handle that.